src/components/advanced-table/advanced-table.tsx

```diff
@@ -153,8 +157,35 @@ export function AdvancedTable<
     setMaxPage(newPage);
   };
 
+  const onLastPage = page === maxPage && nextToken === undefined;
+
   const height = props.height ?? 'auto';
 
+  /**
+   * Renders the label to the left of the pagination buttons.
+   */
+  const labelDisplayedRows = useCallback(
+    ({ from, to, count }: LabelDisplayedRowsArgs) => {
+      if (count === -1) {
+        const loadedRows = rows?.length ?? 0;
+        if (onLastPage) {
+          // for some reason `to` is set incorrectly on the last page in
+          // server-side pagination, so we need to build the string differently
+          // in this case.
+          return trans.__('%1–%2 of %3', from, loadedRows, loadedRows);
+        } else {
+          return (
+            trans.__('%1–%2 of %3', from, to, loadedRows) +
+            (nextToken === undefined ? '' : '+')
```

Comment:
Localized strings shouldn't be concatenated with other strings, even for one character. This covers languages like Japanese where the `loadedRows` parameter wouldn't appear last.

```suggestion
            trans.__('%1–%2 of %3', from, to,
              loadedRows + (nextToken === undefined ? '' : '+'))
```